components/mediation-admin/org.wso2.carbon.mediation.security/src/main/java/org/wso2/carbon/mediation/security/vault/FileSecretRepository.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.mediation.security.vault;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.synapse.SynapseException;
+import org.wso2.securevault.DecryptionProvider;
+import org.wso2.securevault.secret.SecretRepository;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLConnection;
+import java.util.Properties;
+
+/**
+ * Implementation of file based secret repository
+ */
+public class FileSecretRepository implements SecretRepository {
+
+    private static Log LOG = LogFactory.getLog(FileSecretRepository.class);
+
+    /* Parent secret repository */
+    private SecretRepository parentRepository;
+
+    public FileSecretRepository() {
+        super();
+    }
+
+    @Override
+    public void init(Properties properties, String s) {
```

Comment:
Shall we use a proper variable name here?

Reply:
Fixed, thanks